src/controllers/user.controller.ts

```diff
@@ -401,23 +401,58 @@ export const leaveGroup = async (req: Request, res: Response) => {
       $pull: { members: userId },
     });
 
-    // Check if the group exists
     if (targetGroup === null) {
+      // Check if the group exists
       return res.status(400).send({ message: 'Group does not exist!' });
     }
 
-    // Remove groupId from currentGroup of user
-    const targetUser = await User.findByIdAndUpdate(userId, {
-      currentGroup: undefined,
-    });
+    /*
+     * If there are less than 2 members left in the group, delete the group
+     * Else, remove groupId from currentGroup of user
+     */
+    if (targetGroup?.members.length <= 2) {
```

Comment:
shouldn't this be < 2?

Reply:
Nope! Because if the group length is two, then when the person leaves, the length will be one. When the length is one the group should disband.